hibernate-core/src/test/java/org/hibernate/orm/test/idgen/userdefined/UserDefinedGeneratorsTests.java

```diff
@@ -61,10 +62,16 @@ public void testCreateGeneratorsByBeanContainer() {
 		
 		final BeanContainer beanContainer = Mockito.mock( BeanContainer.class );
 		given(beanContainer.getBean( any(), any(), any() ) ).willAnswer( invocation -> {
+			Class<?> beanType = (Class<?>) invocation.getArguments()[0];
 			LifecycleOptions options = (LifecycleOptions) invocation.getArguments()[1];
-			assertThat( options.canUseCachedReferences(), is( false ) );
-			assertThat( options.useJpaCompliantCreation(), is( true ) );
-			return (ContainedBean<?>) TestIdentifierGenerator::new;
+			if (beanType == TestIdentifierGenerator.class) {
```

Comment:
Why are these changes needed?

Reply:
It will fail new tests.

Reply:
It fails because of the changes, this test was not robust enough.